scalabel/label/to_nuscenes.py

```diff
@@ -0,0 +1,292 @@
+    )
+    return parser.parse_args()
+
+
+def to_nuscenes(
+    dataset: Dataset,
+    mode: str,
+) -> DictStrAny:
+    """Conver Scalabel format prediction into nuScenes JSON file."""
+    results: DictStrAny = {}
+
+    assert dataset.groups is not None
+    key_frames = [g for g in dataset.groups if not "_" in g.name]
+
+    if mode == "detection":
+        categories = DETECTION_NAMES
+    else:
+        categories = tracking_cats
+
+    for frame in key_frames:
+        annos = []
+        token = frame.name
+
+        if frame.frameIndex == 0:
+            prev_loc: Dict[str, NDArrayF64] = {}
+        if frame.labels is not None:
+            for label in frame.labels:
+                assert label.category is not None
+                if not label.category in categories:
+                    continue
+
+                assert frame.extrinsics is not None
+                sensor2global = R.from_euler(
+                    "xyz", frame.extrinsics.rotation
+                ).as_matrix()
+
+                assert label.box3d is not None
+                translation = np.dot(
+                    sensor2global, label.box3d.location
+                ) + np.array(frame.extrinsics.location)
+
+                quat = Quaternion(
+                    axis=[0, 1, 0], radians=label.box3d.orientation[1]
+                )
+                x, y, z, w = R.from_matrix(sensor2global).as_quat()
+                rotation = Quaternion([w, x, y, z]) * quat
+
+                h, w, l = label.box3d.dimension
+                dimension = [w, l, h]
+                if mode == "detection":
+                    dimension = [d if d >= 0 else 0.1 for d in dimension]
+
+                tracking_id = label.id
+                if prev_loc.get(tracking_id) is not None:
+                    velocity = translation - prev_loc[tracking_id]
+                else:
+                    velocity = [0.0, 0.0, 0.0]
+                prev_loc[tracking_id] = translation
+
+                if mode == "detection":
+                    nusc_anno = {
+                        "sample_token": token,
+                        "translation": translation.tolist(),
+                        "size": dimension,
+                        "rotation": rotation.elements.tolist(),
+                        "velocity": [velocity[0], velocity[1]],
+                        "detection_name": label.category,
+                        "detection_score": label.score,
+                        "attribute_name": max(
+                            cls_attr_dist[label.category].items(),
+                            key=operator.itemgetter(1),
+                        )[0],
+                    }
+                else:
+                    nusc_anno = {
+                        "sample_token": token,
+                        "translation": translation.tolist(),
+                        "size": dimension,
+                        "rotation": rotation.elements.tolist(),
+                        "velocity": [0.0, 0.0],
```

Comment:
why have no velocity in tracking mode?

Reply:
I recall that it doesn't do anything to the final score. I can fix that.